File: src/main/kotlin/com/github/luoyemyy/mall/common/aspect/AppApi.java
package com.github.luoyemyy.mall.common.aspect;

import java.lang.annotation.*;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface AppApi {

    int pathId();// manager 10000+; applet 20000+;

    boolean auth() default true;
}

File: src/main/kotlin/com/github/luoyemyy/mall/common/aspect/AppApi.kt
package com.github.luoyemyy.mall.common.aspect

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class AppApi(
        val pathId: Int,                // manager 10000+; applet 20000+;
        val pathName: String = "",
        val groupId: Int = 0,
        val groupName: String = "",
        val auth: Boolean = true)